data_steward/cdr_cleaner/cleaning_rules/drop_orphaned_pids.py

```diff
@@ -0,0 +1,173 @@
+SELECT
+  DISTINCT person_id
+FROM
+  `{{project}}.{{dataset}}.{{person_table}}` p
+LEFT JOIN ( SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.condition_occurrence`) co
+USING (person_id)
+LEFT JOIN ( SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.death`) d
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.device_exposure`) dee
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.drug_exposure`) dre
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.measurement`) m
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.note`) n
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.observation`) o
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.procedure_occurrence`) po
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.specimen`) s
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.visit_occurrence`) vo
+USING (person_id)
+LEFT JOIN (SELECT DISTINCT person_id FROM `{{project}}.{{dataset}}.visit_detail`) vd
+USING (person_id)
+where co.person_id is NULL AND d.person_id is NULL AND dee.person_id is NULL AND dre.person_id is NULL 
+AND m.person_id is NULL AND n.person_id is NULL AND o.person_id is NULL AND po.person_id is NULL
+AND s.person_id is NULL AND vo.person_id is NULL AND vd.person_id is NULL
+""")
+
+REMOVE_ORPHANED_PIDS = JINJA_ENV.from_string("""
+DELETE FROM
+  `{{project}}.{{dataset}}.{{person_table}}`
+WHERE
+person_id
+IN ( SELECT
+    person_id
+    FROM `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` )
+""")
+
+
+class DropOrphanedPIDS(BaseCleaningRule):
+    """
+    sandbox and drop participants from the person table if they do not have records in any of the
+    AoU required OMOP tables except person
+    """
+
+    def __init__(self,
+                 project_id,
+                 dataset_id,
+                 sandbox_dataset_id,
+                 table_namer=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Removes all but the most recent of each Physical Measurement for all participants.'
+        super().__init__(issue_numbers=ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[
+                             cdr_consts.COMBINED,
+                             cdr_consts.CONTROLLED_TIER_DEID,
+                             cdr_consts.REGISTERED_TIER_DEID
+                         ],
+                         affected_tables=[PERSON],
+                         project_id=project_id,
```

Comment:
For a moment, I thought we should add `depends_on=DropParticipantsWithoutPPI`. But I realized it is not true for the CT and RT datasets. We can keep it without `depends_on`.